<launch>

  <arg name="onnx_file" default="$(find traffic_light_fine_detector)/data/yolov3-tlr.onnx"/>
  <arg name="mode" default="fp32"/>
  <arg name="read_cache" default="true"/>
  <arg name="input/image" default="/image_raw"/>
  <arg name="input/rois" default="/traffic_light_map_based_detector/output/rois" />
  <arg name="output/rois" default="output/rois" />
  <arg name="score_thresh" default="0.7"/>
  <arg name="approximate_sync" default="false"/>
  <arg name="save_rough_roi_image" default="false"/>
  <node pkg="traffic_light_fine_detector" type="traffic_light_fine_detector" name="traffic_light_fine_detector" output="screen">
    <remap from="~input/image" to="$(arg input/image)" />
    <remap from="~input/rois" to="$(arg input/rois)" />
    <remap from="~output/rois" to="$(arg output/rois)" />
    <param name="onnx_file" type="str" value="$(arg onnx_file)"/>
    <param name="mode" type="str" value="$(arg mode)"/>
    <param name="read_cache" value="$(arg read_cache)"/>
    <param name="score_thresh" value="$(arg score_thresh)"/>
    <param name="approximate_sync" value="$(arg approximate_sync)"/>
  </node>

  <node if="$(eval save_rough_roi_image)" pkg="traffic_light_fine_detector" type="traffic_light_roi_image_saver_node" name="$(anon traffic_light_roi_image_saver)" output="screen">
    <remap from="~input/image" to="$(arg input/image)"/>
    <remap from="~input/rois" to="$(arg input/rois)"/>
    <param name="save_dir" value="$(find traffic_light_fine_detector)/train_image"/>
    <param name="save_rate" value="1.0"/>
  </node>
</launch>
